### PR TITLE
Add HTTP to Core Curriculum

### PR DESCRIPTION
**Pull request by eamelink**
Pretty basic, but should cover the Junior stuff.

### CODE REVIEW

File: core-curriculum/knowledge/http.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "id": "http",
+  "name": "HTTP",
+  "description": "The Hypertext Transfer Protocol, the transport layer of web application.",
+  "tags": [],
+  "topics": [
+    {
+      "id": "http1-basics",
+      "name": "HTTP Basics",
+      "description": "HTTP is the language between browsers or other clients and web applications. Designing web applications involves thinking about the requests that a client will make, and debugging often starts at looking at HTTP requests and responses. Understanding the structure of an HTTP request and response is vital. Dexterity with a tool to create requests and inspect responses is important for every web developer.",
```
Comment (vijaykiran) [inline core-curriculum/knowledge/http.json]: s/language/protocol ?

File: core-curriculum/knowledge/http.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "description": "HTTP statuscodes are standardized and grouped. Knowing the most common status codes and the grouping helps interpret and design restful APIs",
+      "tags": ["required-for-junior"],
+      "dependencies": ["http1-basics"],
+      "resources": [{
+        "name": "HTTP/1.1: Status Code Definitions",
+        "type": "documentation",
+        "url": "https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html"}],
+      "abilities": [],
+      "assessment-questions": [
+        { "question": "What's a 200 response? 403? 500?" },
```
Comment (vijaykiran) [inline core-curriculum/knowledge/http.json]: Perhaps move them into different questions?

File: core-curriculum/knowledge/http.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        { "question": "What the class of 3xx respnses? And 4xx? 5xx?" },
```
Comment (vijaykiran) [inline core-curriculum/knowledge/http.json]: What 'is' the class 

Suggestion - "what does 3xx series response code imply?"

**Comment by eamelink** [comment]
All good suggestions @vijaykiran, all accepted.

**Pull request merged**